### PR TITLE
[Docs] Add Twig Extensions examples

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
follow up to #7531

### CODE REVIEW

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+  
+In addition to the standard Twig extensions, Pimcore adds some additional powerful extensions. 
+  
+## Pimcore Twig Extensions
+
+All Twig extension functions are described below in detail, the following tables give just a short overview of all available extensions.
+
+| Extension                                | Description                                                     |
+|------------------------------------------|-----------------------------------------------------------------|
+| `pimcore_action()`        | (Deprecated) Call an arbitrary action and renders the respective template  |
```
Comment (jdreesen) [inline doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:30]:
```suggestion
| `pimcore_action()`        | (Deprecated) Calls an arbitrary action and renders the respective template  |
```

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+| `pimcore_cache()`         | Simple in-template caching functionality                                         |
+| `pimcore_device()`        | Helps implementing adaptive designs.                                             |
```
Comment (jdreesen) [inline doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:32]:
```suggestion
| `pimcore_device()`        | Helps implementing adaptive designs                                              |
```

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+| `pimcoreglossary()`       | Extension to control the glossary engine                                         |
+| `pimcore_placeholder()`   | Adding and embedding custom placeholders, e.g. for special header tags, etc.     |
+| `pimcore_head_link()`     | Embeding / managing referenced stylesheets (alternative to `assets()`)           |
+| `pimcore_head_meta()`     | Managing your \<meta\> elements in your HTML document                            |
+| `pimcore_head_script()`   | Managing your <scripts> elements                                                 |
+| `pimcore_head_style()`    | Managing inline styles (pendant to `headLink()` for inline styles)               |
+| `pimcore_head_title()`    | Create and store the HTML document's `<title>` for later retrieval and output    |
+| `pimcore_inc()`           | Use this function to directly include a Pimcore document.                        |
```
Comment (jdreesen) [inline doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:40]:
```suggestion
| `pimcore_inc()`           | Use this function to directly include a Pimcore document                         |
```

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+| Extension                                | Description                                                     |
+|------------------------------------------|-----------------------------------------------------------------|
+| `pimcore_action()`        | (Deprecated) Call an arbitrary action and renders the respective template  |
+| `pimcore_cache()`         | Simple in-template caching functionality                                         |
+| `pimcore_device()`        | Helps implementing adaptive designs.                                             |
+| `pimcoreglossary()`       | Extension to control the glossary engine                                         |
+| `pimcore_placeholder()`   | Adding and embedding custom placeholders, e.g. for special header tags, etc.     |
+| `pimcore_head_link()`     | Embeding / managing referenced stylesheets (alternative to `assets()`)           |
+| `pimcore_head_meta()`     | Managing your \<meta\> elements in your HTML document                            |
+| `pimcore_head_script()`   | Managing your <scripts> elements                                                 |
```
Comment (jdreesen) [inline doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:37]:
```suggestion
| `pimcore_head_script()`   | Managing your \<scripts\> elements                                               |
```

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+| `pimcore_head_style()`    | Managing inline styles (pendant to `headLink()` for inline styles)               |
+| `pimcore_head_title()`    | Create and store the HTML document's `<title>` for later retrieval and output    |
+| `pimcore_inc()`           | Use this function to directly include a Pimcore document.                        |
+| `pimcore_inline_script`   | Managing inline scripts (pendant to `headScript()` for inline scripts)           |
+| `pimcore_build_nav()`, `pimcore_render_nav()`, `pimcore_nav_renderer()`   | Embed and build navigations based on the document structure                             |
+| `pimcore_url()`           | An alternative to `url()` and `path()` with the building behavior of Pimcore 4   |
```
Comment (jdreesen) [inline doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:43]: `with the building behavior of Pimcore 4` is this relevant?

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+| `pimcore_website_config()`| Fetch website settings or specific setting(first param: key) for the current site |
```
Comment (jdreesen) [inline doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:44]:
```suggestion
| `pimcore_website_config()`| Fetch website settings or specific setting (first param: key) for the current site |
```

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+| `pimcore_image_thumbnail()` | Returns a path to a given thumbnail on image                                   |
+| `pimcore_image_thumbnail_html()` | Returns html for displaying the thumbnail image.                          |
```
Comment (jdreesen) [inline doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:46]:
```suggestion
| `pimcore_image_thumbnail_html()` | Returns html for displaying the thumbnail image                           |
```

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md [hunk 8]
```diff
@@ -0,0 +1,288 @@
...
+| `pimcore_supported_locales()` | Use this function to get list of support locales.                            |
```
Comment (jdreesen) [inline doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:47]:
```suggestion
| `pimcore_supported_locales()` | Use this function to get a list of supported locales                         |
```

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+| `pimcore_image_thumbnail()` | Returns a path to a given thumbnail or an thumbnail configuration.             |
```
Comment (jdreesen) [inline doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:48]:
```suggestion
| `pimcore_image_thumbnail()` | Returns a path to a given thumbnail or thumbnail configuration                 |
```

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md [hunk 10]
```diff
@@ -0,0 +1,288 @@
...
+| `pimcore_image_thumbnail()` | Returns a path to a given thumbnail or an thumbnail configuration.             |
```
Comment (jdreesen) [inline doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:49]:
```suggestion
| `pimcore_image_thumbnail()` | Returns a path to a given thumbnail or thumbnail configuration                 |
```

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md [hunk 11]
```diff
@@ -0,0 +1,288 @@
...
+| `pimcore_head_meta()`     | Managing your \<meta\> elements in your HTML document                            |
+| `pimcore_head_script()`   | Managing your <scripts> elements                                                 |
+| `pimcore_head_style()`    | Managing inline styles (pendant to `headLink()` for inline styles)               |
+| `pimcore_head_title()`    | Create and store the HTML document's `<title>` for later retrieval and output    |
+| `pimcore_inc()`           | Use this function to directly include a Pimcore document.                        |
+| `pimcore_inline_script`   | Managing inline scripts (pendant to `headScript()` for inline scripts)           |
+| `pimcore_build_nav()`, `pimcore_render_nav()`, `pimcore_nav_renderer()`   | Embed and build navigations based on the document structure                             |
+| `pimcore_url()`           | An alternative to `url()` and `path()` with the building behavior of Pimcore 4   |
+| `pimcore_website_config()`| Fetch website settings or specific setting(first param: key) for the current site |
+| `pimcore_image_thumbnail()` | Returns a path to a given thumbnail on image                                   |
```
Comment (jdreesen) [inline doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:45]:
```suggestion
| `pimcore_image_thumbnail()` | Returns a path to a given thumbnail or image                                   |
```

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md [hunk 12]
```diff
@@ -0,0 +1,288 @@
...
+
+Pimcore also adds some Twig tests for evaluating boolean conditions e.g.
+```twig
+{# using 'instaceof' checks if object is instanceof provided classname #}
+{% if( product is instanceof('AppBundle\\Model\\Product\\Car')) %}
+    ...
+{% endif %}
+
+{# using 'pimcore_data_object' checks if object is instanceof \Pimcore\Model\DataObject\Concrete #}
+{% if( product is pimcore_data_object) %}
```
Comment (jdreesen) [inline doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:59]:
```suggestion
{% if (product is pimcore_data_object) %}
```

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md [hunk 13]
```diff
@@ -0,0 +1,288 @@
...
+ ...
+{% endif %}
+```
+
+The following tables give an overview of all available tests:
```
Comment (jdreesen) [inline doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:64]:
```suggestion
The following table gives an overview of all available tests:
```

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md [hunk 14]
```diff
@@ -0,0 +1,288 @@
...
+### `pimcore_inc` 
+Use `pimcore_inc()` to include documents (eg. snippets) within views. 
+This is especially useful for footers, headers, navigations, sidebars, teasers, ...
+
+`pimcore_inc(document, params, cacheEnabled)`
+
+| Name           | Type         | Description  |
+|----------------|--------------|--------------|
+| `document`     | PageSnippet &#124; int &#124; string  | Document to include, can be either an ID, a path or even the Document object itself |
+| `params`       |  array       | Is optional and should be an array with key value pairs like in `$this->action()` from ZF. |
```
Comment (jdreesen) [inline doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:195]: `...from ZF.` there is no Zend anymore ;)

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md [hunk 15]
```diff
@@ -0,0 +1,288 @@
...
+| `enabledCache` |  bool        | Is true by default, set it to false to disable the cache. Hashing is done across source and parameters to ensure a consistent result. |
+ 
+ ##### Example
```
Comment (jdreesen) [inline doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:198]:
```suggestion
##### Example
```

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md [hunk 16]
```diff
@@ -0,0 +1,288 @@
...
+
+| Name         | Type   | Description  |
+|--------------|--------|--------------|
+| `params`     | array  | Route params. If object is passed in the params then link generator will be used to generate Url  |
+| `name`       | string | Route name |
+| `reset`      | bool   | Is false by default, set it to false to avoid merging parameters from request  |
+| `encode`     | bool   | Is true by default, set it to false to disable encoding |
+| `relative`   | bool   | Is false by default, set it to true to generate a relative path based on the current request path |
+
+ ##### Example
```
Comment (jdreesen) [inline doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:284]:
```suggestion
##### Example
```

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md (hunk 11 above)
Comment (dvesh3) [reply doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:45]: this is called on image

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md (hunk 5 above)
Comment (dvesh3) [reply doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:43]: kicked it

File: doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md (hunk 14 above)
Comment (dvesh3) [reply doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md:195]: kicked it

thanks for your review 🙏

**Pull request merged**